src/server/common_services/auth_server.js

```diff
@@ -543,7 +542,8 @@ function _prepare_auth_request(req) {
         }
 
         // check that auth contains valid system role or the account is support
-        if (!ignore_missing_system) {
+        // We should not check for roles and accounts in anonymous access
+        if (!ignore_missing_system && !allow_anonymous_access) {
```

Comment:
There's a nuance here that we want to be strict about which is when the method allows anonymous, but the request is authenticated, in this case I think it is best to check the request as authenticated and verify the roles are valid, so the condition should just check for preconditions to the request having the account and system info -
```suggestion
        if (req.auth && req.system && req.account) {
```